clusterbuster

```diff
@@ -2351,6 +2354,91 @@ EOF
     done
 }
 
+function create_vm_deployment() {
+    local namespace=$1
+    local instance=$2
+    local -i replicas=$4
+    local -i replica=0
+    local depname="${namespace}-${workload}-${instance}"
+    while (( replica++ < replicas )) ; do
+        local name="${depname}-${replica}"
+        local vmname=$(mkpodname "$name")
+        create_object -n "$namespace" "$name" <<EOF
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  labels:
+    app: $name
+    kubevirt-vm: $vmname
+  name: $vmname
+$(indent 2 standard_deployment_metadata_yaml "$namespace" client)
+$(indent 2 standard_pod_metadata_yaml "$namespace" client)
+$(create_vm_spec -n "$vmname")
+EOF
+    done
+}
+
+function create_vm_spec() {
+    local image
+    local vmname
+    local cores
+    local sockets
+    local threads
+    local mem
+    local grace
+    local OPT
+    local OPTIND=1
+    local OPTARG
+    while getopts ':n:c:s:t:m:g:i:-' OPT; do
+        case "$OPT" in
+        '?') fatal "Unknown option -$OPTARG" ;;
+         : ) fatal "Option -$OPTARG requires a value" ;;
+         - ) break ;; # end options with --
+         n ) vmname=$OPTARG ;;
+         c ) cores=$OPTARG ;;
+         s ) sockets=$OPTARG ;;
+         t ) threads=$OPTARG ;;
+         m ) mem=$OPTARG ;;
+         g ) grace=$OPTARG ;;
+         i ) image=$OPTARG ;;
+         * ) fatal "Unhandled option -$OPT"
+        esac
+    done
+    shift $((OPTIND-1))
+    [[ -z $vmname ]] && fatal "vmname is required"
```

Comment:
Better:

[[ -n "$vmname" ]] || fatal "vmname is required"

Aside from needing to quote the vmname in case it contains spaces, the construct you use will always have a value of false, which would cause failure if we use the -e option.  It should have a value true in the expected case.